### PR TITLE
Add Rails 7 support

### LINKED ISSUE #523: Support for Rails 7
Rails 7 has been released and it would be great if cucumber-rails could support it. 

I did an initial "fix" where I just relaxed the gemspec requirement and added test runners for 7.0 in https://github.com/erikaxel/cucumber-rails/commit/6b225611fad8af12780f3718bd9f9c63327c4462
However that seems to be insufficient since a couple of tests fail. I'm not very familiar with this project so I'm a bit stumped on where to continue. Happy to help if I can, but it would be great with some pointers on where to start. The errors on the CI can be seen here: https://github.com/erikaxel/cucumber-rails/actions/runs/1587333272
 
PS: Also happy to make my initial commit with tests running on 7.0 into a PR if that is helpful.

### PR DESCRIPTION
**Pull request by mgrunberg**
## Summary

Fixes #523

The PR was built based on issue comments.

Three tests were failing. Two because Rails 7 uses modern HTML control (input type "datetime-local") for datepickers. Capybara helper acts differently based on Rails version

The other test was failing because Destroy link is not present on index action anymore. The test changed to navigate the corresponding page based on Rails version.

### CODE REVIEW

**Comment by tagliala** [comment]
Hi, I can see that [specs are failing](https://github.com/mgrunberg/cucumber-rails/runs/4824503004?check_suite_focus=true) against Ruby 2.5 but it looks like an issue not related to this change

**Comment by mgrunberg** [comment]
Yes, the issue is not related to rails 7 changes. Made another PR that doesn't change the code and [ruby 2.5 fails too](https://github.com/mgrunberg/cucumber-rails/runs/4824473170?check_suite_focus=true).

I'm looking for solutions / explanations. Maybe drop ruby 2.5 support is an alternative.

**Comment by tagliala** [comment]
```
  webdrivers was resolved to 4.6.1, which depends on
    selenium-webdriver was resolved to 3.142.7, which depends on
      childprocess
```

maybe it requires an older version of webdrivers, selenium-webdriver, or childprocess. I wasn't able to replicate on my development machine, so I'm not sure

**Comment by mgrunberg** [comment]
From what I can tell is an issue of the environment (ruby + bundler + rubygems) because bundler fails fetching.
Fixed the problem by running `gem update --system`
Just with that [all test pass](https://github.com/mgrunberg/cucumber-rails/actions/runs/1702129170)

**Comment by luke-hill** [comment]
Yes, I found an issue of webdrivers compatibility on another gem I support. See here: https://github.com/site-prism/site_prism/blob/main/gemfiles/Gemfile.low_spec#L9

File: cucumber-rails.gemspec [hunk 1]
```diff
@@ -25,7 +25,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency('cucumber', ['>= 3.2', '< 8']) # Support cucumber in the 3.x <-> 7.x revision range
   s.add_runtime_dependency('mime-types', ['~> 3.3']) # Only support the latest major (3+ years old)
   s.add_runtime_dependency('nokogiri', '~> 1.10') # Only support the latest major (3+ years old)
-  s.add_runtime_dependency('railties', ['>= 5.0', '< 7']) # We support any version of Rails in the 5.x and 6.x series
+  s.add_runtime_dependency('railties', ['>= 5.0', '< 7.1']) # We support any version of Rails in the 5.x, 6.x and 7.0.x series
```
Comment (luke-hill) [inline cucumber-rails.gemspec:28]: This should be permissible for all 7.x series gems.

File: features/emulate_javascript.feature [hunk 2]
```diff
@@ -88,7 +89,19 @@ Feature: Emulate Javascript
...
       When('I follow {string}') do |link|
-        click_link(link)
+        if ::Rails::VERSION::MAJOR >= 7
+          click_button(link)
+        else
+          click_link(link)
+        end
+      end
+
+      When('I am on destroy {string} page') do |name|
```
Comment (luke-hill) [inline features/emulate_javascript.feature:99]: can we change this to be a Given step

**Review by luke-hill — COMMENTED** [review]
Couple of minor tweaks

**Review by luke-hill — APPROVED** [review]

**Comment by mgrunberg** [comment]
@luke-hill I asked for review and then saw your approval, sorry.

**Comment by tagliala** [comment]
Thanks!

**Comment by jeygeethan** [comment]
Is this version released?

**Comment by luke-hill** [comment]
We're currently merging in lots of items. A release will be cut when appropriate.

**Comment by julitrows** [comment]
Hi, thanks for the hard work on getting the gem RoR7-ready! ❤️ 

Any ETA on the new version being cut and released? 😬 

Thanks a heap 🙇

**Comment by luke-hill** [comment]
Honestly we don't sorry. 

I will commit to it being out in the next month, I appreciate that's not a great answer. Just a few things going on atm.

**Comment by luke-hill** [comment]
https://rubygems.org/gems/cucumber-rails/versions/2.5.0 is now released.

**Comment by mathieujobin** [comment]
Thanks a lot @luke-hill 

Just in time for the sakura 桜🌸

**Pull request merged**